src/GitVersionCore/IncrementStrategyFinder.cs

```diff
@@ -80,14 +81,28 @@ public static class IncrementStrategyFinder
         
         private static IEnumerable<Commit> GetIntermediateCommits(IRepository repo, Commit baseCommit, Commit headCommit)
         {
-            var filter = new CommitFilter
+            if (baseCommit == null) yield break;
+
+            if (intermediateCommitCache == null || intermediateCommitCache.LastOrDefault() != headCommit)
+            {
+                var filter = new CommitFilter
+                {
+                    Since = headCommit,
+                    SortBy = CommitSortStrategies.Topological | CommitSortStrategies.Reverse
+                };
+
+                intermediateCommitCache = repo.Commits.QueryBy(filter).ToList();
+            }
+
+            var found = false;
+            foreach (var commit in intermediateCommitCache)
             {
-                Since = headCommit,
-                Until = baseCommit,
-                SortBy = CommitSortStrategies.Topological | CommitSortStrategies.Reverse
-            };
+                if (commit.Sha == baseCommit.Sha)
```

Comment:
@JakeGinnivan why are we comparing the `Sha` instead of the commit? We're using `Sha` for comparison over the whole code base so I'll guess it's the way to go, but I'm just curious about why?